File: drivers/java/src/test/java/mesh/SmokeTest.java
// Smoke tests — proves the handshake + a single RUN/PULL roundtrip
// work via the official neo4j-java-driver. Mirrors Python's
// test_smoke.py, JS's smoke.test.js, and Go's smoke_test.go.

package mesh;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;
import org.neo4j.driver.Driver;
import org.neo4j.driver.Result;
import org.neo4j.driver.Session;

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class SmokeTest {
    private Driver driver;

    @BeforeAll
    void openDriver() {
        driver = Connection.openDriver();
    }

    @AfterAll
    void closeDriver() {
        driver.close();
    }

    @Test
    void returnOne() {
        try (Session session = driver.session()) {
            Result result = session.run("RETURN 1 AS n");
            assertEquals(1L, result.single().get("n").asLong());
        }
    }

    @Test
    void helloCompletes() {
        // If the handshake or HELLO phase failed, the session/run
        // would throw. This test just asserts we get a summary
        // back — makes a regression bisect's top-level failure
        // point explicit.
        try (Session session = driver.session()) {
            Result result = session.run("RETURN 1");
            assertNotNull(result.consume());
        }
    }
}

File: drivers/js/test/smoke.test.js
// Smoke test — proves the handshake + a single RUN/PULL roundtrip
// work via the official Neo4j JS driver. Mirrors Python's test_smoke.py.

import { after, before, test } from 'node:test';
import assert from 'node:assert/strict';

import { openDriver } from './helpers.js';

let driver;

before(() => {
  driver = openDriver();
});

after(async () => {
  await driver.close();
});

test('RETURN 1 round-trips', async () => {
  const session = driver.session();
  try {
    const result = await session.run('RETURN 1 AS n');
    assert.equal(result.records.length, 1);
    // Integer handling: neo4j-driver wraps ints in a BigInt-ish class
    // by default to preserve 64-bit precision. `toNumber` is the
    // documented way to recover a JS number when the value fits in
    // IEEE 754 double precision.
    assert.equal(result.records[0].get('n').toNumber(), 1);
  } finally {
    await session.close();
  }
});

test('HELLO completed — metadata comes back on consume()', async () => {
  // If the handshake or HELLO phase failed, the session.run above
  // would throw. This test just reasserts a successful consume() to
  // make regression bisects easier to read.
  const session = driver.session();
  try {
    const summary = await session.run('RETURN 1').then((r) => r.summary);
    assert.ok(summary !== null);
  } finally {
    await session.close();
  }
});
